### PR TITLE
Add changeset to modules

### PR DESCRIPTION
**Pull request by ykostov**
WIP, do not merge

### CODE REVIEW

File: lib/bgsite_official/categories/tag.ex [hunk 1]
```diff
@@ -17,4 +17,20 @@ defmodule BgsiteOfficial.Categories.Tag do
     |> cast(attrs, [:title, :banner, :description, :likes])
     |> validate_required([:title, :banner, :description, :likes])
   end
+
+  def changeset_update_projects(%Tag{} = tags, attrs) do
```
Comment (rmilushev) [inline lib/bgsite_official/categories/tag.ex:21]: I think the name should not include `projects`, but something like `changeset_update_tags_for_websites`.

File: lib/bgsite_official/categories/tag.ex [hunk 2]
```diff
@@ -17,4 +17,20 @@ defmodule BgsiteOfficial.Categories.Tag do
...
+      |> cast(%{}, @required_fields)
+      # associate projects to the user
+      |> put_assoc(:websites, websites)
+  end
+
+  many_to_many
+  (
+      :websites,
+      Website,
+      join_through: "website_project",
```
Comment (rmilushev) [inline lib/bgsite_official/categories/tag.ex:32]: The join table (when created) will be `website_tag`, not `.._project`. We will generate a migration for the join table.

File: lib/bgsite_official/categories/tag.ex [hunk 3]
```diff
@@ -17,4 +17,20 @@ defmodule BgsiteOfficial.Categories.Tag do
...
+      website
+      |> cast(%{}, @required_fields)
+      # associate projects to the user
+      |> put_assoc(:websites, websites)
+  end
+
+  many_to_many
+  (
+      :websites,
+      Website,
```
Comment (rmilushev) [inline lib/bgsite_official/categories/tag.ex:31]: The [module name](https://github.com/N-Patarov/bgsite_official/pull/20/files) is `Websites`.

**Review by rmilushev — CHANGES_REQUESTED** [review]
We have to generate a migration for the join table `website_tag` like this:
```
mix ecto.gen.migration add_website_tag_table websites:references tag:references
```

**Comment by ykostov** [comment]
:+1:

**Pull request closed without merging**